dev-libs/xxhash/xxhash-0.6.4.recipe

```diff
@@ -0,0 +1,64 @@
+SUMMARY="A fast non-cryptographic hash algorithm"
+DESCRIPTION="xxHash is an extremely fast non-cryptographic hash algorithm, \
+working at speeds close to RAM limits."
+HOMEPAGE="https://cyan4973.github.io/xxHash/
+	http://www.xxhash.com/"
+COPYRIGHT="2012-2016 Yann Collet"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="http://github.com/Cyan4973/xxHash/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="4570ccd111df6b6386502791397906bf69b7371eb209af7d41debc2f074cdb22"
+SOURCE_FILENAME="xxHash-$portVersion.tar.gz"
+SOURCE_DIR="xxHash-$portVersion"
+PATCHES="xxHash-$portVersion.patch"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	libxxhash$secondaryArchSuffix = $portVersion
+	cmd:xxh32sum$secondaryArchSuffix = $portVersion
+	cmd:xxh64sum$secondaryArchSuffix = $portVersion
+	cmd:xxhsum$secondaryArchSuffix = $portVersion
+	lib:libxxhash$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libxxhash${secondaryArchSuffix}_devel = $portVersion
+	devel:libxxhash$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libxxhash$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage libxxhash$secondaryArchSuffix \
+	"$binDir"/xxh32sum "$binDir"/xxh64sum "$binDir"/xxhsum \
```

Comment:
```diff
 defineDebugInfoPackage libxxhash$secondaryArchSuffix \
-	"$binDir"/xxh32sum "$binDir"/xxh64sum "$binDir"/xxhsum \
+	"$binDir"/xxhsum \
-	"$libDir/libxxhash.so.$libVersion"
+	"$libDir"/libxxhash.so.$libVersion
```
(xxh32sum and xxh64sum are just symlinks, so they can't be passed to `defineDebugInfoPackage`.)
For the line with the shared lib, I think the closing quotes look nicer after the `libDir`, but this is just my point of view, so please feel free to keep it where you prefer :)

BTW, the recipe builds fine, but I'm trying to see if we can change it a little to:
* add a `TEST()` with `make test-all`.
* see if it builds on x86_gcc2 with a few changes in the CFLAGS.
* if it does not build on x86_gcc2, use the [`commandBinDir`](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#using-the-commandbindir-user-variable) trick.

Reply:
Inserting a `PATCH()` function after the last line of the call to `defineDebugInfoPackage` and before the `BUILD()` allows to build for x86_gcc2 :)
```bash
PATCH()
{
	if [ "$effectiveTargetArchitecture" = x86_gcc2 ]; then
		sed -i \
			-e "s/ -Wextra//" \
			-e "s/ -Wstrict-aliasing=1//" \
			-e "s/ -Wswitch-enum//" \
			-e "s/ -Wdeclaration-after-statement//" \
			Makefile
	fi
}
```
(We like to have an empty line before and after this bloc, like in 8dd83364dfc08075d5ac885ec61e7325361c0180.)

So, since it builds on all architectures, we can probably forget about the commandBinDir trick.
Well, if the tests are all OK on x86_gcc2.
If not, then we should:
1) Build the lib for x86_gcc2 primary arch and drop the cmd:xxhash on that arch.
2) Use the commandBinDir and keep the xxhash built for x86 secondary arch so that it can be used on x86_gcc2.

This is not simple the first time you see this, but I'll look for a package with a similar layout and explain it better.

Reply:
Fixed.